src/Identifier/TokenIdentifier.php

```diff
@@ -12,34 +12,32 @@
  */
 namespace Authentication\Identifier;
 
-use Cake\ORM\TableRegistry;
-use InvalidArgumentException;
-use RuntimeException;
+use Authentication\Identifier\Backend\DatasourceAwareTrait;
 
 /**
  * Token Identifier
  */
 class TokenIdentifier extends AbstractIdentifier
 {
 
+    use DatasourceAwareTrait;
+
     /**
-     * Default configuration
+     * Default configuration.
      *
      * @var array
      */
     protected $_defaultConfig = [
         'tokenField' => 'token',
         'dataField' => 'token',
-        'userModel' => 'Users',
-        'finder' => 'all',
-        'tokenVerification' => 'Orm'
+        'datasource' => 'Authentication.Orm'
     ];
 
     /**
-     * Identify
+     * Identify method.
      *
-     * @param array $data Authentication credentials
-     * @return \Cake\Datasource\EntityInterface|null
+     * @param array $data Authentication credentials.
+     * @return \ArrayAccess|null
```

Comment:
@burzum right. 
`DatasourceInterface` returns `ArrayAccess` in my case because we can't force users to return EntityInterface in their implementations I think. So I changed the typehint  here. I could revert this though but let's hear what others have to say about it.